dev_tools/ci/size-labeler.sh

```diff
@@ -0,0 +1,213 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail -o errtrace
+shopt -s inherit_errexit
+
+declare -r usage="Usage: ${0##*/} [-h | --help | help]
+
+Updates the size labels on a pull request based on the number of lines it
+changes.  The script requires the following environment variables:
+PR_NUMBER, GITHUB_REPOSITORY, GITHUB_TOKEN.  The script is intended
+for automated execution from GitHub Actions workflow."
+
+declare -ar LABELS=(
+    "Size: XS"
+    "size: S"
+    "size: M"
+    "size: L"
+    "size: XL"
+)
+
+declare -A LIMITS=(
+    ["${LABELS[0]}"]=10
+    ["${LABELS[1]}"]=50
+    ["${LABELS[2]}"]=200
+    ["${LABELS[3]}"]=800
+    ["${LABELS[4]}"]="$((2 ** 63 - 1))"
+)
+
+declare -ar IGNORED=(
+    "*_pb2.py"
+    "*_pb2.pyi"
+    "*_pb2_grpc.py"
+    ".*.lock"
+    "*.bundle.js"
+)
+
+function info() {
+    echo >&2 "INFO: ${*}"
+}
+
+function error() {
+    echo >&2 "ERROR: ${*}"
+}
+
+function jq_stdin() {
+    local infile
+    infile="$(mktemp)"
+    readonly infile
+
+    cat >"${infile}"
+    jq_file "$@" "${infile}"
+}
+
+function jq_file() {
+    # Regardless of the success, store the return code.
+    # Prepend each sttderr with command args and send back to stderr.
+    jq "${@}" 2> >(awk -v h="stderr from jq ${*}:" '{print h, $0}' 1>&2) &&
+        rc="${?}" ||
+        rc="${?}"
+    if [[ "${rc}" != "0" ]]; then
+        error "The jq program failed: ${*}"
+        error "Note the quotes above may be wrong. Here was the (possibly empty) input in ${*: -1}:"
+        cat "${@: -1}" # Assumes last argument is input file!!
+    fi
+    return "${rc}"
+}
+
+function api_call() {
+    local -r endpoint="${1// /%20}" # love that our label names have spaces...
+    local -r uri="https://api.github.com/repos/${GITHUB_REPOSITORY}"
+    local -r url="${uri}/${endpoint}"
+    local -r curl_opts=(
+        -fsSL
+        --connect-timeout 10 --max-time 20
+        -H "Authorization: token ${GITHUB_TOKEN}" \
+        -H "Accept: application/vnd.github.v3.json" \
+        -H "X-GitHub-Api-Version:2022-11-28" \
+        -H "Content-Type: application/json" \
+    )
+    info "Calling: ${url}"
+    set +e
+    response_body=$(curl "${curl_opts[@]}" "${@:2}" "${url}")
+    local exit_status=$?
+    set -e
+    echo "${response_body}"
+    if [[ $exit_status -ne 0 ]]; then
+        error "GitHub API call failed (curl exit $exit_status) for ${url}"
+        exit $exit_status
+    fi
+}
+
+function compute_changes() {
+    local -r pr="$1"
+
+    local change_info
+    local -r keys_filter='with_entries(select([.key] | inside(["changes", "filename"])))'
+    change_info="$(jq_stdin "map(${keys_filter})" <<<"$(api_call "pulls/${pr}/files")")"
```

Comment:
Is this based on an older version of the Cirq size-labeler.sh?

I recall fixing these constructs (`cmd <<<"$(api_call ...)"`) which ignore the exit status of the api_call and run `cmd` for cirq.

Please consider syncing this with the current Cirq script which is known to work.